### ISSUE TITLE
testing-library/prefer-screen-queries with self-written helpers

### ISSUE THREAD

**Issue by pustovalov** [testing-library/eslint-plugin-testing-library#340]
Hey

```
expect(await findByComplexText(screen, 'Selected campaigns 0')).toBeInTheDocument()
```

`eslint-plugin-testing-library@3.10.2` - no errors

`eslint-plugin-testing-library@4.0.1` - Avoid destructuring queries from `render` result, use `screen.findByComplexText` instead  testing-library/prefer-screen-queries

should this rule only apply to `testing-library` methods?

**Comment by Belco90** [comment]
Hello @pustovalov. Did you check [the migration guide to v4](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/migrating-to-v4-guide.md)? That error is reported by `prefer-screen-queries` which has been enabled in all Shareable Configs and mentioned as one of the things to be aware of when migrating.

You are still able to switch the rule off if you are not interested in it tough.

**Comment by Belco90** [comment]
My bad, I think I misunderstood the actual issue.

You are getting a non-related Testing Library method reported as a Testing Library query right? Can you confirm if your `findByComplexText` is a custom query? If so, I think we need to improve the checks in this rule to avoid reporting custom ones since they are not attached to `screen` (I'm not sure if this is possible actually).

If it's not a custom query but something completely unrelated to Testing Library, Aggressive Reporting is causing that but there is no option to switch it off for queries yet. It should be available in the following days.

**Comment by pustovalov** [comment]
>That error is reported by prefer-screen-queries which has been enabled in all Shareable Configs and mentioned as one of the things to be aware of when migrating.

I have used this rule before and there were no errors

this is not custom query, it's just a function which uses `screen`, this is a `findByComplexText`:


```
export function findByComplexText(screen, text) {
  return screen.findByText((content, node) => {
    const hasText = node => node.textContent === text
    const nodeHasText = hasText(node)
    const childrenDontHaveText = Array.from(node.children).every(child => !hasText(child))

    return nodeHasText && childrenDontHaveText
  })
}
```

**Comment by Belco90** [comment]
Thanks for sharing! That function is understood by the plugin as a custom query, which is fine! So you can get the advantage of other rules linting that function, like `await-async-query` complaining about not awaiting your `findByComplexText` for example.

However, it doesn't make sense to get `prefer-screen-queries` reporting custom queries since you are not able to access them through `screen`. This is a bug indeed, but should be easy to solve! Thanks for reporting and sorry about the misunderstood in my first message.

We'll get back when it's fixed 🚀

**Comment by pustovalov** [comment]
@Belco90 Thx, new version works fine
